core/src/main/java/com/orientechnologies/orient/core/serialization/serializer/record/binary/ORecordSerializerBinaryV1.java

```diff
@@ -824,124 +1112,120 @@ private static void writeSBTreeRidbag(BytesContainer bytes, ORidBag ridbag, UUID
 
     OVarIntSerializer.write(bytes, pointer.getFileId());
     OVarIntSerializer.write(bytes, pointer.getRootPointer().getPageIndex());
     OVarIntSerializer.write(bytes, pointer.getRootPointer().getPageOffset());
     OVarIntSerializer.write(bytes, ridbag.size());
 
-    if (context != null){
-      ((OSBTreeRidBag)ridbag.getDelegate()).handleContextSBTree(context, pointer);
+    if (context != null) {
+      ((OSBTreeRidBag) ridbag.getDelegate()).handleContextSBTree(context, pointer);
       OVarIntSerializer.write(bytes, 0);
-    }
-    else{
+    } else {
       OVarIntSerializer.write(bytes, 0);
 
       //removed changes serialization
     }
   }
-  
-  private static int writeRidBag(BytesContainer bytes, ORidBag ridbag) {    
+
+  private static int writeRidBag(BytesContainer bytes, ORidBag ridbag) {
     ridbag.checkAndConvert();
-    
+
     UUID ownerUuid = ridbag.getTemporaryId();
-    
+
     int positionOffset = bytes.offset;
     final OSBTreeCollectionManager sbTreeCollectionManager = ODatabaseRecordThreadLocal.instance().get()
         .getSbTreeCollectionManager();
     UUID uuid = null;
     if (sbTreeCollectionManager != null)
       uuid = sbTreeCollectionManager.listenForChanges(ridbag);
-    
+
     byte configByte = 0;
     if (ridbag.isEmbedded())
       configByte |= 1;
 
     if (uuid != null)
       configByte |= 2;
-    
+
     //alloc will move offset and do skip
     int posForWrite = bytes.alloc(OByteSerializer.BYTE_SIZE);
-    OByteSerializer.INSTANCE.serialize(configByte, bytes.bytes, posForWrite);    
-    
+    OByteSerializer.INSTANCE.serialize(configByte, bytes.bytes, posForWrite);
+
     //removed serializing UUID
-    
+
     if (ridbag.isEmbedded()) {
       writeEmbeddedRidbag(bytes, ridbag);
-    } else { 
+    } else {
       writeSBTreeRidbag(bytes, ridbag, ownerUuid);
     }
     return positionOffset;
   }
-  
-  private static ORidBag readRidbag(BytesContainer bytes){
-    byte configByte = OByteSerializer.INSTANCE.deserialize(bytes.bytes, bytes.offset++);    
+
+  private static ORidBag readRidbag(BytesContainer bytes) {
+    byte configByte = OByteSerializer.INSTANCE.deserialize(bytes.bytes, bytes.offset++);
     boolean isEmbedded = (configByte & 1) != 0;
 //    boolean hasUUID = (configByte & 2) != 0;
-    
+
     UUID uuid = null;
     //removed deserializing UUID
-    
+
     ORidBag ridbag = null;
-    if (isEmbedded){
+    if (isEmbedded) {
       ridbag = new ORidBag();
       int size = OVarIntSerializer.readAsInteger(bytes);
       ridbag.getDelegate().setSize(size);
-      for (int i = 0; i < size; i++){
+      for (int i = 0; i < size; i++) {
         OIdentifiable record = readLinkOptimizedEmbedded(bytes);
-        if (record == null){
+        if (record == null) {
           //should never happen
           String errorMessage = "Deserialized null object during ridbag deserialization";
           OSerializationException exc = new OSerializationException("");
```

Comment:
Can you add here the correct message

Reply:
as readLinkOptimizedEmbedded(byte[]) never returns null, this check is redundant, so it is removed